Visual Script Fixes

jdbener opened the pull request:
1) Fixed bug causing variable setters/getters added from the add node menu to be unusable when only one variable is present.
2) Updated setter/gettter port naming to make the node's state more clear.
3) Fixed regression causing script members to not update.

jdbener committed:
Visual Script Fixes

1) Fixed bug causing variable setters/getters added from the add node menu to be unusable when only one variable is present.
2) Updated port naming to make the node's state more clear.
3) Fixed regression causing script members to not update.

swarnimarun commented:
Nice.
But is there a reason for you to have rename _update_members ? I had fixed it on my branch(I am waiting for more breaking changes to get committed before making PRs) and there wasn't a need for the change, something I missed?
Secondly, I would recommend to keep all the method binds inside of `VisualScriptEditor::_bind_methods()` for easier maintainability.

And at last I am not sure why it was required to add another function to the `VisualScriptNode` class.?

swarnimarun commented on modules/visual_script/register_types.cpp:
```diff
@@ -116,6 +116,10 @@ void register_visual_script_types() {
 	ClassDB::register_class<_VisualScriptEditor>();
+
+	ClassDB::register_class<VisualScriptEditor>();
+	ClassDB::bind_method(D_METHOD("update_members"), &VisualScriptEditor::update_members);
```
Move it inside VisualScriptEditor::_bind_methods()

jdbener commented:
> But is there a reason for you to have rename _update_members ? I had fixed it on my branch(I am waiting for more breaking changes to get committed before making PRs) and there wasn't a need for the change, something I missed?

The main reason for the change was consistency, as I understand it functions which start with an underscore usually aren't 'publicly' exposed to the ClassDB. Since it was getting publicly exposed I changed the name to match the convention. Is this incorrect?

jdbener commented:
> And at last I am not sure why it was required to add another function to the `VisualScriptNode` class.?

There wasn't a virtual function which got executed when a node was added to the graph. I could have made `validate_input_default_values()` fit the bill but that seamed obscure.

swarnimarun commented:
@fire, @akien-mga I think we can merge it if everyone is fine with adding a new function to `VisualScriptNode` class.

swarnimarun commented:
I believe,
We could just check which is the first property in the Constructor as well considering VisualScript has to be created before Nodes.
@jdbener would it not have worked?

Otherwise it's fine by me.

swarnimarun commented:
Hmmm. That's something I didn't expect but if that's the case then I believe it's fine to add another function in VisualScriptNode.

akien-mga commented:
This was superseded by #49749, but it would be worth checking if all fixes from this PR are fully superseded or if some parts of it should be put in a new PR.
